src/pages/VaccinePage.js

```diff
@@ -1,16 +1,104 @@
+/* eslint-disable react/forbid-prop-types */
 /**
  * mSupply Mobile
  * Sustainable Solutions (NZ) Ltd. 2016
  */
 
 import React from 'react';
-import { View } from 'react-native';
 import { connect } from 'react-redux';
+import { StyleSheet, Text, FlatList } from 'react-native';
+import PropTypes from 'prop-types';
 
-export const VaccinePageComponent = () => <View />;
+import { FridgeDisplay, FlexView } from '../widgets';
 
-const mapStateToProps = state => state;
+import { selectMinAndMaxLogs, selectMinAndMaxDomains } from '../selectors/fridge';
+import { FridgeActions } from '../actions/FridgeActions';
 
-const mapDispatchToProps = dispatch => ({ dispatch });
+import { APP_FONT_FAMILY } from '../globalStyles';
+import { vaccineStrings } from '../localization';
+
+export const VaccinePageComponent = ({
+  selectedFridge,
+  minLine,
+  maxLine,
+  breaches,
+  maxDomain,
+  minDomain,
+  fridges,
+  onSelectFridge,
+}) => {
+  const Fridge = React.useCallback(
+    ({ item }) => (
+      <FridgeDisplay
+        fridge={item}
+        minLine={minLine}
+        maxLine={maxLine}
+        breaches={breaches}
+        maxDomain={maxDomain}
+        minDomain={minDomain}
+        onSelectFridge={onSelectFridge}
+        isActive={item.id === selectedFridge.id}
+      />
+    ),
+    minLine,
+    maxLine,
+    breaches,
+    maxDomain,
+    minDomain,
+    onSelectFridge
+  );
+
+  const BlankComponent = React.useCallback(
+    () => (
+      <FlexView flex={1} justifyContent="center" alignItems="center">
+        <Text style={localStyles.blankText}>{vaccineStrings.no_fridges}</Text>
+      </FlexView>
+    ),
+    []
+  );
+
+  return fridges.length ? <FlatList data={fridges} renderItem={Fridge} /> : <BlankComponent />;
```

Comment:
Not sure of naming of below suggested component, but I think avoids calculating `fridges.length` on every render here?

 ```suggestion
const FridgesList = useCallback(() => !!fridges.length ? <FlatList data={fridges} renderItem={Fridge} /> : <BlankComponent />, [fridges]);
return <FridgesList />;
```

Reply:
I think there's quite a few comments re caching which we might disagree on. Mainly because it isn't really going to offer any performance benefit and imo actually makes it practically harder to read in my opinion.. gotta look in two places+ to see exactly whats being returned. 

I think "memoize all the things" is a bit of a trap and I'm trying to do it sparingly and with purpose when I do as it can just lead to so many small, subtle and annoying bugs! There doesn't seem to be toooo much benefit in this case